crypto.js

```diff
@@ -18,52 +17,50 @@ var concat = Buffer.concat
 
 var nonce = new Buffer(24); nonce.fill(0)
 
-var challenge_length = 64
-var client_auth_length = 16+32+64
-var server_auth_length = 16+64
-var mac_length = 16
+var isBuffer = Buffer.isBuffer
 
-//this is a simple secure handshake,
-//the client public key is passed in plain text,
+exports.challenge_length = 64
+exports.client_auth_length = 16+32+64
+exports.server_auth_length = 16+64
+exports.mac_length = 16
 
-module.exports = State
+//both client and server
 
-function State (app_key, local, remote, seed) {
+exports.initialize = function (state) {
 
-  if(!(this instanceof State)) return new State(app_key, local, remote, seed)
+  if(state.seed) state.local = from_seed(state.seed)
 
-  if(seed) local = from_seed(seed)
+  //TODO: sodium is missing box_seed_keypair. should make PR for that.
 
-  this.app_key = app_key
-  var kx = keypair()
-  this.local = {
-    kx_pk: kx.publicKey,
-    kx_sk: kx.secretKey,
-    public: local.publicKey,
-    secret: local.secretKey
-  }
-  this.remote = {
-    public: remote || null
-  }
+  var _key = from_seed(state.random)
+//  var kx = keypair(random)
+  var kx_pk = curvify_pk(_key.publicKey)
+  var kx_sk = curvify_sk(_key.secretKey)
 
-}
+  state.local = {
+    kx_pk: kx_pk,
+    kx_sk: kx_sk,
+    publicKey: state.local.publicKey,
+    secretKey: state.local.secretKey,
+    app_mac: auth(kx_pk, state.app_key)
+  }
 
-var proto = State.prototype
+  state.local.kx_pk = kx_pk
+  state.local.kx_sk = kx_sk
+  state.local.app_mac = auth(kx_pk, state.app_key)
```

Comment:
I am not very good at javascript but aren't these assignments redundant? `state.local` has been initialized several lines above: `kx_pk`, `kx_sk` and `app_mac` already have those values.